### PR TITLE
Enum database value customization

### PR DESCRIPTION
**Pull request by stevenschlansker**
Fixes #1561

### CODE REVIEW

File: core/src/main/java/org/jdbi/v3/core/enums/DatabaseValue.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.jdbi.v3.core.qualifier.Qualifier;
+import org.jdbi.v3.meta.Beta;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
+@Qualifier
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/core/enums/DatabaseValue.java]: This shouldn't be a qualifier annotation, should it?

File: core/src/main/java/org/jdbi/v3/core/mapper/EnumMapper.java [hunk 2]
```diff
@@ -78,6 +81,11 @@ public E map(ResultSet rs, int columnNumber, StatementContext ctx) throws SQLExc
         private static Object getValueByName(Class<? extends Enum<?>> enumClass, String name) {
             final Enum<?>[] enumConstants = enumClass.getEnumConstants();
             return JdbiOptionals.findFirstPresent(
+                    () -> Arrays.stream(enumConstants).filter(e -> {
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/core/mapper/EnumMapper.java]: If somebody annotates an enum with `@DatabaseValue` then I don't think we should be falling back to matching the enum name.

File: core/src/main/java/org/jdbi/v3/core/enums/DatabaseValue.java (hunk 1 above)
Comment (maffe) [reply core/src/main/java/org/jdbi/v3/core/enums/DatabaseValue.java]: I don’t know what `@Qualifier` does, it’s an oversight from copy & paste.

Comment (qualidafial) [reply core/src/main/java/org/jdbi/v3/core/enums/DatabaseValue.java]: Qualifying annotations are a recent addition to Jdbi, typically used to distinguish when multiple SQL data types resolve to the same Java type. e.g. both `VARCHAR` and `NVARCHAR` SQL columns are represented as `String` in Java.

http://jdbi.org/#_qualified_types

Comment (stevenschlansker) [reply core/src/main/java/org/jdbi/v3/core/enums/DatabaseValue.java]: https://github.com/jdbi/jdbi/commit/c6e243cc4c374fe3510b3b3a967bc4c8f6e31cdc

File: core/src/main/java/org/jdbi/v3/core/mapper/EnumMapper.java (hunk 2 above)
Comment (stevenschlansker) [reply core/src/main/java/org/jdbi/v3/core/mapper/EnumMapper.java]: https://github.com/jdbi/jdbi/pull/1566

Comment (maffe) [reply core/src/main/java/org/jdbi/v3/core/mapper/EnumMapper.java]: > If somebody annotates an enum with `@DatabaseValue` then I don't think we should be falling back to matching the enum name.

If I understand you correctly, given the already included enum
```
private enum Foobar {
    BAR, FOO, @DatabaseValue("CUST") CUSTOM
}
```
this test should pass (currently it doesn’t) and be included:
```
@Test
public void customizedNamesAreNotMappedToEnumName() throws NoSuchFieldException {
    db.getJdbi().useHandle(h -> {
        Optional<Foobar> mappedEnum = h.createQuery("select :name")
                .bind("name", "CUSTOM")
                .mapTo(Foobar.class)
                .findOne();

        assertThat(mappedEnum)
                .isEmpty();
    });
}
```

**Pull request merged**